malaria24/ona/tasks.py

```diff
@@ -46,18 +46,18 @@ def ona_fetch_reported_case_for_form(form_id):
 
         case = ReportedCase.objects.create(
             form=OnaForm.objects.get(form_id=form_id),
-            first_name=data['first_name'],
-            last_name=data['last_name'],
-            locality=data['locality'],
-            date_of_birth=data['date_of_birth'],
-            create_date_time=data['create_date_time'],
+            first_name=data.get('first_name') or "",
+            last_name=data.get('last_name') or "",
+            locality=data.get('locality') or '_other',
+            date_of_birth=data.get('date_of_birth'),
+            create_date_time=data.get('create_date_time'),
             sa_id_number=data.get('sa_id_number'),
-            msisdn=data['msisdn'],
-            id_type=data['id_type'],
-            abroad=data['abroad'],
-            reported_by=data['reported_by'],
-            gender=data['gender'],
-            facility_code=data['facility_code'],
+            msisdn=data.get('msisdn'),
```

Comment:
Some of these fields are required so will fail if `.get()` returns `None`

Reply:
Happy to leave this as is for now and let it error